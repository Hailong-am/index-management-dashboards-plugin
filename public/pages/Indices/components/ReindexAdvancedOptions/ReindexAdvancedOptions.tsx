/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { ChangeEvent } from "react";
import CustomFormRow from "../../../../components/CustomFormRow";
import {
  EuiComboBox,
  EuiComboBoxOptionOption,
  EuiFieldText,
  EuiLink,
  EuiRadioGroup,
  EuiSpacer,
  EuiSwitch,
  EuiSwitchEvent,
  EuiText,
} from "@elastic/eui";
import { DSL_DOCUMENTATION_URL } from "../../../../utils/constants";
import JSONEditor from "../../../../components/JSONEditor";

interface ReindexOptionsProps {
  slices: string;
  onSlicesChange: (e: ChangeEvent<HTMLInputElement>) => void;
  sliceErr?: string;
  pipelines: EuiComboBoxOptionOption<void>[];
  selectedPipelines?: EuiComboBoxOptionOption<void>[];
  onSelectedPipelinesChange: (options: EuiComboBoxOptionOption<void>[]) => void;
  queryJsonString: string;
  onQueryJsonChange: (val: string) => void;
  conflicts: string;
  onConflictsChange: (val: string) => void;
  subset: boolean;
  onSubsetChange: (event: EuiSwitchEvent) => void;
}

const ReindexAdvancedOptions = (props: ReindexOptionsProps) => {
  const {
    slices,
    sliceErr,
    onSlicesChange,
    pipelines,
    selectedPipelines,
    onSelectedPipelinesChange,
    queryJsonString,
    onQueryJsonChange,
  } = props;
  const { conflicts, onConflictsChange } = props;
  const { subset, onSubsetChange } = props;

  return (
    <div style={{ padding: "10px 10px" }}>
      <CustomFormRow label="Reindex subset of source">
        <EuiSwitch
          label="Reindex subset of source"
          data-test-subj="subsetSwitch"
          showLabel={false}
          compressed
          checked={subset}
          onChange={onSubsetChange}
        />
      </CustomFormRow>

      {subset && (
        <CustomFormRow
          label="Query expression"
          labelAppend={
            <EuiText size="xs">
              <EuiLink href={DSL_DOCUMENTATION_URL} target="_blank" rel="noopener noreferrer">
                learn more about query-dsl
              </EuiLink>
            </EuiText>
          }
        >
          <JSONEditor
            mode="json"
            width="100%"
            value={queryJsonString}
            onChange={onQueryJsonChange}
            aria-label="Query DSL Editor"
            height="150px"
            data-test-subj="queryJsonEditor"
          />
        </CustomFormRow>
      )}

      <CustomFormRow
        label="How to handle conflicts"
        fullWidth={true}
        helpText="Set to proceed to continue reindexing even if there are conflicts. Defaults to abort"
      >
        <EuiRadioGroup
          options={[
            {
              id: "proceed",
              label: "proceed",
            },
            {
              id: "abort",
              label: "abort",
            },
          ]}
          idSelected={conflicts}
          onChange={onConflictsChange}
          name="conflicts group"
        />
      </CustomFormRow>

      <EuiSpacer size="m" />

      <CustomFormRow
        isInvalid={!!sliceErr}
        error={sliceErr}
        fullWidth={true}
        label="Slices"
        helpText="Number of sub-tasks OpenSearch should divide this task into. Default is 1, which means OpenSearch should not divide this task. Setting this parameter to auto indicates to OpenSearch that it should automatically decide how many slices to split the task into."
      >
        <EuiFieldText data-test-subj="slices" value={slices} onChange={onSlicesChange} />
      </CustomFormRow>

      <EuiSpacer size="m" />
      <CustomFormRow label="Pipeline" helpText="Pipeline pre-process documents before writing into destination">
        <EuiComboBox
          aria-label="Ingest Pipeline"
          placeholder="Select a single pipeline"
          data-test-subj="pipelineCombobox"
          singleSelection={{ asPlainText: true }}
          options={pipelines}
          selectedOptions={selectedPipelines}
          onChange={onSelectedPipelinesChange}
        />
      </CustomFormRow>
    </div>
  );
};

export default ReindexAdvancedOptions;
