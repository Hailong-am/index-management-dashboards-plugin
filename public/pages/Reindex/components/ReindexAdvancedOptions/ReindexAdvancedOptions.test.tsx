/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from "@testing-library/react";
import React from "react";
import ReindexAdvancedOptions from "./ReindexAdvancedOptions";

describe("<ReindexAdvancedOptions /> spec", () => {
  it("renders the component", async () => {
    let component = render(
      <ReindexAdvancedOptions
        slices="1"
        onSlicesChange={() => {}}
        pipelines={[]}
        selectedPipelines={[]}
        onSelectedPipelinesChange={() => {}}
        conflicts="proceed"
        onConflictsChange={() => {}}
        onQueryJsonChange={() => {}}
        onSubsetChange={() => {}}
        queryJsonString="{}"
        subset={true}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders the component with slice error", async () => {
    const { getByText } = render(
      <ReindexAdvancedOptions
        slices="1"
        onSlicesChange={() => {}}
        sliceErr={"slice must be positive integer or auto"}
        pipelines={[]}
        selectedPipelines={[]}
        onSelectedPipelinesChange={() => {}}
        conflicts="proceed"
        onConflictsChange={() => {}}
        onQueryJsonChange={() => {}}
        onSubsetChange={() => {}}
        queryJsonString="{}"
        subset={false}
      />
    );

    expect(getByText("slice must be positive integer or auto")).toBeInTheDocument();
  });
});
