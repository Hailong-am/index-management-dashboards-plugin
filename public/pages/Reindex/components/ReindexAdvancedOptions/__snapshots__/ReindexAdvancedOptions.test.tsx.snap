// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReindexAdvancedOptions /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="padding: 10px 10px;"
      >
        <div
          class="euiFormRow"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              Reindex subset of source
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiSwitch euiSwitch--compressed"
            >
              <button
                aria-checked="true"
                aria-label="Reindex subset of source"
                class="euiSwitch__button"
                data-test-subj="subsetSwitch"
                id="some_html_id"
                role="switch"
                type="button"
              >
                <span
                  class="euiSwitch__body"
                >
                  <span
                    class="euiSwitch__thumb"
                  />
                  <span
                    class="euiSwitch__track"
                  />
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFormRow"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              Query expression
            </label>
             
            <div
              class="euiText euiText--extraSmall"
            >
              <a
                class="euiLink euiLink--primary"
                href="https://opensearch.org/docs/opensearch/query-dsl/index/"
                rel="noopener noreferrer"
                target="_blank"
              >
                learn more about query-dsl
                EuiIconMock
                <span
                  class="euiScreenReaderOnly"
                >
                  (opens in a new tab or window)
                </span>
              </a>
            </div>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <textarea
              data-test-subj="queryJsonEditor-value-display"
              readonly=""
              style="display: none;"
            >
              {}
            </textarea>
            <div
              class="euiCodeEditorWrapper"
              data-test-subj="queryJsonEditor"
              style="width: 100%; height: 150px;"
            >
              <button
                class="euiCodeEditorKeyboardHint"
                data-test-subj="codeEditorHint"
                id="some_html_id"
              >
                <p
                  class="euiText"
                >
                  Press Enter to start editing.
                </p>
                <p
                  class="euiText"
                >
                  When you're done, press Escape to stop editing.
                </p>
              </button>
              <div
                class=" ace_editor ace-tm"
                id="some_html_id"
                style="width: 100%; height: 150px;"
              >
                <textarea
                  aria-label="Query DSL Editor"
                  autocapitalize="off"
                  autocorrect="off"
                  class="ace_text-input"
                  spellcheck="false"
                  style="opacity: 0;"
                  tabindex="-1"
                  wrap="off"
                />
                <div
                  aria-hidden="true"
                  class="ace_gutter"
                >
                  <div
                    class="ace_layer ace_gutter-layer ace_folding-enabled"
                  />
                  <div
                    class="ace_gutter-active-line"
                  />
                </div>
                <div
                  class="ace_scroller"
                >
                  <div
                    class="ace_content"
                  >
                    <div
                      class="ace_layer ace_print-margin-layer"
                    >
                      <div
                        class="ace_print-margin"
                        style="left: 4px; visibility: visible;"
                      />
                    </div>
                    <div
                      class="ace_layer ace_marker-layer"
                    />
                    <div
                      class="ace_layer ace_text-layer"
                      style="padding: 0px 4px;"
                    />
                    <div
                      class="ace_layer ace_marker-layer"
                    />
                    <div
                      class="ace_layer ace_cursor-layer ace_hidden-cursors"
                    >
                      <div
                        class="ace_cursor"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="ace_scrollbar ace_scrollbar-v"
                  style="display: none; width: 20px;"
                >
                  <div
                    class="ace_scrollbar-inner"
                    style="width: 20px;"
                  />
                </div>
                <div
                  class="ace_scrollbar ace_scrollbar-h"
                  style="display: none; height: 20px;"
                >
                  <div
                    class="ace_scrollbar-inner"
                    style="height: 20px;"
                  />
                </div>
                <div
                  style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
                >
                  <div
                    style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                  />
                  <div
                    style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                  >
                    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              How to handle conflicts
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormHelpText euiFormRow__text"
              style="padding-top: 0px; padding-bottom: 4px;"
            >
              Set to proceed to continue reindexing even if there are conflicts. Defaults to abort
            </div>
            <div>
              <div
                class="euiRadio euiRadioGroup__item"
              >
                <input
                  checked=""
                  class="euiRadio__input"
                  id="proceed"
                  name="conflicts group"
                  type="radio"
                  value=""
                />
                <div
                  class="euiRadio__circle"
                />
                <label
                  class="euiRadio__label"
                  for="proceed"
                >
                  proceed
                </label>
              </div>
              <div
                class="euiRadio euiRadioGroup__item"
              >
                <input
                  class="euiRadio__input"
                  id="abort"
                  name="conflicts group"
                  type="radio"
                  value=""
                />
                <div
                  class="euiRadio__circle"
                />
                <label
                  class="euiRadio__label"
                  for="abort"
                >
                  abort
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid="false"
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              Slices
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormHelpText euiFormRow__text"
              style="padding-top: 0px; padding-bottom: 4px;"
            >
              Number of sub-tasks OpenSearch should divide this task into. Default is 1, which means OpenSearch should not divide this task. Setting this parameter to auto indicates to OpenSearch that it should automatically decide how many slices to split the task into.
            </div>
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  class="euiFieldText"
                  data-test-subj="slices"
                  type="text"
                  value="1"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiFormRow"
          id="some_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="some_html_id"
            >
              Pipeline
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormHelpText euiFormRow__text"
              style="padding-top: 0px; padding-bottom: 4px;"
            >
              Pipeline pre-process documents before writing into destination
            </div>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="Ingest Pipeline"
              class="euiComboBox"
              data-test-subj="pipelineCombobox"
              role="combobox"
            >
              <div
                class="euiFormControlLayout"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <div
                    class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                    data-test-subj="comboBoxInput"
                    tabindex="-1"
                  >
                    <p
                      class="euiComboBoxPlaceholder"
                    >
                      Select a single pipeline
                    </p>
                    <div
                      class="euiComboBox__input"
                      style="font-size: 14px; display: inline-block;"
                    >
                      <input
                        aria-controls=""
                        data-test-subj="comboBoxSearchInput"
                        role="textbox"
                        style="box-sizing: content-box; width: 2px;"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                  <div
                    class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                  >
                    <button
                      aria-label="Open list of options"
                      class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                      data-test-subj="comboBoxToggleListButton"
                      type="button"
                    >
                      EuiIconMock
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="padding: 10px 10px;"
    >
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Reindex subset of source
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiSwitch euiSwitch--compressed"
          >
            <button
              aria-checked="true"
              aria-label="Reindex subset of source"
              class="euiSwitch__button"
              data-test-subj="subsetSwitch"
              id="some_html_id"
              role="switch"
              type="button"
            >
              <span
                class="euiSwitch__body"
              >
                <span
                  class="euiSwitch__thumb"
                />
                <span
                  class="euiSwitch__track"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Query expression
          </label>
           
          <div
            class="euiText euiText--extraSmall"
          >
            <a
              class="euiLink euiLink--primary"
              href="https://opensearch.org/docs/opensearch/query-dsl/index/"
              rel="noopener noreferrer"
              target="_blank"
            >
              learn more about query-dsl
              EuiIconMock
              <span
                class="euiScreenReaderOnly"
              >
                (opens in a new tab or window)
              </span>
            </a>
          </div>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <textarea
            data-test-subj="queryJsonEditor-value-display"
            readonly=""
            style="display: none;"
          >
            {}
          </textarea>
          <div
            class="euiCodeEditorWrapper"
            data-test-subj="queryJsonEditor"
            style="width: 100%; height: 150px;"
          >
            <button
              class="euiCodeEditorKeyboardHint"
              data-test-subj="codeEditorHint"
              id="some_html_id"
            >
              <p
                class="euiText"
              >
                Press Enter to start editing.
              </p>
              <p
                class="euiText"
              >
                When you're done, press Escape to stop editing.
              </p>
            </button>
            <div
              class=" ace_editor ace-tm"
              id="some_html_id"
              style="width: 100%; height: 150px;"
            >
              <textarea
                aria-label="Query DSL Editor"
                autocapitalize="off"
                autocorrect="off"
                class="ace_text-input"
                spellcheck="false"
                style="opacity: 0;"
                tabindex="-1"
                wrap="off"
              />
              <div
                aria-hidden="true"
                class="ace_gutter"
              >
                <div
                  class="ace_layer ace_gutter-layer ace_folding-enabled"
                />
                <div
                  class="ace_gutter-active-line"
                />
              </div>
              <div
                class="ace_scroller"
              >
                <div
                  class="ace_content"
                >
                  <div
                    class="ace_layer ace_print-margin-layer"
                  >
                    <div
                      class="ace_print-margin"
                      style="left: 4px; visibility: visible;"
                    />
                  </div>
                  <div
                    class="ace_layer ace_marker-layer"
                  />
                  <div
                    class="ace_layer ace_text-layer"
                    style="padding: 0px 4px;"
                  />
                  <div
                    class="ace_layer ace_marker-layer"
                  />
                  <div
                    class="ace_layer ace_cursor-layer ace_hidden-cursors"
                  >
                    <div
                      class="ace_cursor"
                    />
                  </div>
                </div>
              </div>
              <div
                class="ace_scrollbar ace_scrollbar-v"
                style="display: none; width: 20px;"
              >
                <div
                  class="ace_scrollbar-inner"
                  style="width: 20px;"
                />
              </div>
              <div
                class="ace_scrollbar ace_scrollbar-h"
                style="display: none; height: 20px;"
              >
                <div
                  class="ace_scrollbar-inner"
                  style="height: 20px;"
                />
              </div>
              <div
                style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
              >
                <div
                  style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                />
                <div
                  style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                >
                  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFormRow euiFormRow--fullWidth"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            How to handle conflicts
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            Set to proceed to continue reindexing even if there are conflicts. Defaults to abort
          </div>
          <div>
            <div
              class="euiRadio euiRadioGroup__item"
            >
              <input
                checked=""
                class="euiRadio__input"
                id="proceed"
                name="conflicts group"
                type="radio"
                value=""
              />
              <div
                class="euiRadio__circle"
              />
              <label
                class="euiRadio__label"
                for="proceed"
              >
                proceed
              </label>
            </div>
            <div
              class="euiRadio euiRadioGroup__item"
            >
              <input
                class="euiRadio__input"
                id="abort"
                name="conflicts group"
                type="radio"
                value=""
              />
              <div
                class="euiRadio__circle"
              />
              <label
                class="euiRadio__label"
                for="abort"
              >
                abort
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiFormRow euiFormRow--fullWidth"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Slices
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            Number of sub-tasks OpenSearch should divide this task into. Default is 1, which means OpenSearch should not divide this task. Setting this parameter to auto indicates to OpenSearch that it should automatically decide how many slices to split the task into.
          </div>
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldText"
                data-test-subj="slices"
                type="text"
                value="1"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Pipeline
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            Pipeline pre-process documents before writing into destination
          </div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Ingest Pipeline"
            class="euiComboBox"
            data-test-subj="pipelineCombobox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Select a single pipeline
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      aria-controls=""
                      data-test-subj="comboBoxSearchInput"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
                <div
                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <button
                    aria-label="Open list of options"
                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                    data-test-subj="comboBoxToggleListButton"
                    type="button"
                  >
                    EuiIconMock
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
